[gym] Add `export_xcargs` option

nubbel opened the pull request:
Thanks for contributing to _fastlane_! Before you submit your pull request, please make sure to check the following boxes:

- [x] Run `bundle exec rspec` from the subdirectory of each tool you modified. Alternatively, run `rake test_all` from the root directory.
- [x] Run `bundle exec rubocop -a` to ensure the code style is valid
- [x] Read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] We currently don't accept new actions, please publish a plugin instead, more information in [Plugins.md](https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Plugins.md)

Before submitting a pull request, we appreciate if you create an issue first to discuss the change :+1:

fixes #6761

Commit:
[gym] Add `export_xcargs` option

fixes #6761

milch commented on gym/lib/gym/generators/package_command_generator_xcode7.rb:
```diff
@@ -22,15 +22,14 @@ def generate
-        if Gym.config[:toolchain]
-          options << "-toolchain '#{Gym.config[:toolchain]}'"
-        end
+        options << "-toolchain '#{Gym.config[:toolchain]}'" if Gym.config[:toolchain]
```
Cleaning up the code as you go, I love it 😍  Although you did create the `config` var above, so it might be worthwhile to replace `Gym.config` with that.

nubbel replied:
Sorry, that was my intention in the first place. I'll change that quickly. Thanks!

milch commented on gym/lib/gym/options.rb:
```diff
@@ -123,6 +123,14 @@ def self.plain_options
                                      end),
+        FastlaneCore::ConfigItem.new(key: :export_xcargs,
+                                     env_name: "GYM_EXPORT_XCARGS",
+                                     description: "Pass additional arguments to xcodebuild for the export phase. Be sure to quote the setting names and values e.g. OTHER_LDFLAGS=\"-ObjC -lstdc++\"",
```
Please also update the regular `xcargs` action description to reflect that that only applies to the build phase.

nubbel replied:
Good point. Thanks!

milch reviewed (changes requested):
Looking good! Just some tiny changes needed before we can merge 🚀

nubbel commented:
@milch Thanks for your feedback. I've updated the PR with the changes you suggested.
Regarding the option descriptions, I changed the wording to "build" and "package" phase such that it is on par with the command generators.

Cheers 🍻 (🍼 might be more appropriate though)

milch commented:
@nubbel Thanks for taking the time to contribute by fixing this issue! Yup, that wording sounds good. 

Cheers! 🍻  

(Also, next time to make it easier on us we'd prefer if you made a separate commit to address the review issues instead of `--amend`ing. That makes it easier for us to review the second time as we only have to look at the changes)

nubbel commented:
@milch Thanks for merging!

> (Also, next time to make it easier on us we'd prefer if you made a separate commit to address the review issues instead of --amending. That makes it easier for us to review the second time as we only have to look at the changes)

Sorry about that, I'm addicted to force push 😜. I wanted to keep the history clean and forgot that Github can now squash commits before merge. Will keep that in mind for my next PR.

milch commented:
Yeah no worries, it happens all the time - the "Squash and Merge" feature is still pretty new after all. It's totally fine for smaller PRs like this one, though!

KrauseFx commented:
This is a great addition 👍
